HACKING.md

```diff
@@ -14,3 +16,106 @@ echo "Ending pre-commit"
 ````
 
 into the `.git/hooks/pre-commit` file.
+
+#### Makefile
```

Comment:
We don't need to document the `Makefile` here. It is better to say what do you need to make after a modification, i.e. `make tests`, `make hlint` and `make check-whitespace`.